src/jwks.cc

```diff
@@ -365,12 +380,42 @@ JwksPtr Jwks::createFrom(const std::string& pkey, Type type) {
     case Type::PEM:
       keys->createFromPemCore(pkey);
       break;
+    case Type::PKCS8:
+      keys->createFromPkcs8Core(pkey);
+      break;
     default:
       break;
   }
   return keys;
 }
 
+// pkey_pem must be a PEM-encoded PKCS #8 public key.
+// This is the format that starts with -----BEGIN PUBLIC KEY-----.
+// Currently this only supports RSA. Support for ECC will be added soon.
+void Jwks::createFromPkcs8Core(const std::string& pkey_pem) {
+  keys_.clear();
+  PubkeyPtr key_ptr(new Pubkey());
+  EvpPkeyGetter e;
+  key_ptr->evp_pkey_ = e.createEvpPkeyFromPkcs8(pkey_pem);
+  updateStatus(e.getStatus());
+  if (key_ptr->evp_pkey_ == nullptr) {
+    assert(e.getStatus() != Status::Ok);
+    return;
+  }
+
+  key_ptr->pem_format_ = true;
+  switch (EVP_PKEY_type(key_ptr->evp_pkey_->type)) {
+    case EVP_PKEY_RSA:
+      key_ptr->alg_ = "RSA";
+      break;
+    default:
+      updateStatus(Status::JwksNotImplementedKty);
```

Comment:
prefer to define a new status specific for PKCS

Reply:
Done